model/riscv_insts_zkn_rv64.sail

```diff
@@ -0,0 +1,166 @@
+
+/*
+ * Scalar Cryptography Extension - Scalar 64-bit AES instructions
+ * ----------------------------------------------------------------------
+ */
+
+union clause ast = AES64KS1I : (bits(4), regidx, regidx)
+union clause ast = AES64KS2  : (regidx, regidx, regidx)
+union clause ast = AES64IM   : (regidx, regidx)
+union clause ast = AES64ESM  : (regidx, regidx, regidx)
+union clause ast = AES64ES   : (regidx, regidx, regidx)
+union clause ast = AES64DSM  : (regidx, regidx, regidx)
+union clause ast = AES64DS   : (regidx, regidx, regidx)
+
+mapping clause encdec = AES64KS1I (rcon, rs1, rd) if sizeof(xlen) == 64
+  <-> 0b00 @ 0b11000 @ 0b1 @ rcon @ rs1 @ 0b001 @ rd @ 0b0010011
+
+mapping clause encdec = AES64IM (rs1, rd) if sizeof(xlen) == 64
+  <-> 0b00 @ 0b11000 @ 0b00000 @ rs1 @ 0b001 @ rd @ 0b0010011
+
+mapping clause encdec = AES64KS2 (rs2, rs1, rd) if sizeof(xlen) == 64
+  <-> 0b01 @ 0b11111 @ rs2 @ rs1 @ 0b000 @ rd @ 0b0110011
+
+mapping clause encdec = AES64ESM (rs2, rs1, rd) if sizeof(xlen) == 64
+  <-> 0b00 @ 0b11011 @ rs2 @ rs1 @ 0b000 @ rd @ 0b0110011
+
+mapping clause encdec = AES64ES (rs2, rs1, rd) if sizeof(xlen) == 64
+  <-> 0b00 @ 0b11001 @ rs2 @ rs1 @ 0b000 @ rd @ 0b0110011
+
+mapping clause encdec = AES64DSM (rs2, rs1, rd) if sizeof(xlen) == 64
+  <-> 0b00 @ 0b11111 @ rs2 @ rs1 @ 0b000 @ rd @ 0b0110011
+
+mapping clause encdec = AES64DS (rs2, rs1, rd) if sizeof(xlen) == 64
+  <-> 0b00 @ 0b11101 @ rs2 @ rs1 @ 0b000 @ rd @ 0b0110011
+
+mapping clause assembly = AES64KS1I (rcon, rs1, rd)
+  <-> "aes64ks1i" ^ spc() ^ reg_name(rd) ^ sep() ^ reg_name(rs1) ^ sep() ^ hex_bits_4(rcon)
+
+mapping clause assembly = AES64KS2 (rs2, rs1, rd)
+  <-> "aes64ks2" ^ spc() ^ reg_name(rd) ^ sep() ^ reg_name(rs1) ^ sep() ^ reg_name(rs2)
+
+mapping clause assembly = AES64IM (rs1, rd) 
+  <-> "aes64im" ^ spc() ^ reg_name(rd) ^ sep() ^ reg_name(rs1)
+
+mapping clause assembly = AES64ESM (rs2, rs1, rd)
+  <-> "aes64esm" ^ spc() ^ reg_name(rd) ^ sep() ^ reg_name(rs1) ^ sep() ^ reg_name(rs2)
+
+mapping clause assembly = AES64ES (rs2, rs1, rd)
+  <-> "aes64es" ^ spc() ^ reg_name(rd) ^ sep() ^ reg_name(rs1) ^ sep() ^ reg_name(rs2)
+
+mapping clause assembly = AES64DSM (rs2, rs1, rd)
+  <-> "aes64dsm" ^ spc() ^ reg_name(rd) ^ sep() ^ reg_name(rs1) ^ sep() ^ reg_name(rs2)
+
+mapping clause assembly = AES64DS (rs2, rs1, rd)
+  <-> "aes64ds" ^ spc() ^ reg_name(rd) ^ sep() ^ reg_name(rs1) ^ sep() ^ reg_name(rs2)
+
+function clause execute (AES64KS1I(rcon, rs1, rd)) = {
+  let tmp1 : bits(32) = X(rs1)[63..32];
+  let rc   : bits(32) = aes_decode_rcon(rcon);
+  let tmp2 : bits(32) = if (rcon ==0xA) then tmp1 else ror32(tmp1, 8);
+  let tmp3 : bits(32) = aes_subword_fwd(tmp2);
+  let result : bits(64) = (tmp3 ^ rc) @ (tmp3 ^ rc);
```

Comment:
I'd have to go back and very carefully study the AES spec and these instructions to remember Barry's explanation of how/why these instructions work as they do. But you are on the right track in suggesting its because they work across all parametrisations of AES. Which again, isn't intuitive from their specification.